consumer/api.go

```diff
@@ -0,0 +1,94 @@
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+)
+
+type ValidatorTracker struct {
+	PeerID            string  `json:"peer_id"`
+	ENR               string  `json:"enr"`
+	Multiaddr         string  `json:"multiaddr"`
+	IP                string  `json:"ip"`
+	Port              int     `json:"port"`
+	LastSeen          int     `json:"last_seen"`
+	LastEpoch         int     `json:"last_epoch"`
+	ClientVersion     string  `json:"client_version"`
+	PossibleValidator bool    `json:"possible_validator"`
+	MaxValidatorCount int     `json:"max_validator_count"`
+	NumObservations   int     `json:"num_observations"`
+	Hostname          string  `json:"hostname"`
+	City              string  `json:"city"`
+	Region            string  `json:"region"`
+	Country           string  `json:"country"`
+	Latitude          float64 `json:"latitude"`
+	Longitude         float64 `json:"longitude"`
+	PostalCode        string  `json:"postal_code"`
+	ASN               string  `json:"asn"`
+	ASNOrganization   string  `json:"asn_organization"`
+	ASNType           string  `json:"asn_type"`
+}
+
+var selectQuery = `SELECT peer_id, enr, multiaddr, validator_tracker.ip, port, last_seen, last_epoch, client_version, possible_validator, max_validator_count, num_observations, hostname, city, region, country, latitude, longitude, postal_code, asn, asn_organization, asn_type FROM validator_tracker JOIN ip_metadata ON validator_tracker.ip = ip_metadata.ip`
+
+// LoadAPIKeys reads the API keys from a file and returns a map of keys
+func LoadAPIKeys(filePath string) (map[string]bool, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	apiKeys := make(map[string]bool)
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		if line != "" {
+			apiKeys[line] = true
+		}
+	}
+	return apiKeys, scanner.Err()
+}
+
+func createGetValidatorsHandler(db *sql.DB, apiKeys map[string]bool) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		apiKey := r.Header.Get("API-Key")
+		isAdmin := apiKeys[apiKey]
+
+		rows, err := db.Query(selectQuery)
+		if err != nil {
+			http.Error(w, "Error querying database", http.StatusInternalServerError)
+			return
+		}
+		defer rows.Close()
+
+		var validators []ValidatorTracker
+		for rows.Next() {
+			var vm ValidatorTracker
+			err := rows.Scan(&vm.PeerID, &vm.ENR, &vm.Multiaddr, &vm.IP, &vm.Port, &vm.LastSeen, &vm.LastEpoch, &vm.ClientVersion, &vm.PossibleValidator, &vm.MaxValidatorCount, &vm.NumObservations, &vm.Hostname, &vm.City, &vm.Region, &vm.Country, &vm.Latitude, &vm.Longitude, &vm.PostalCode, &vm.ASN, &vm.ASNOrganization, &vm.ASNType)
+			if err != nil {
+				http.Error(w, fmt.Sprintf("Error scanning row: %s", err), http.StatusInternalServerError)
+				return
+			}
+
+			// If the user is not an admin, we should not return the sensitive data
+			if !isAdmin {
+				vm.Multiaddr = ""
+				vm.IP = ""
```

Comment:
Let's add ENR to this (ENR also contains the IP address)

Reply:
ah my bad